### DISCUSSION TITLE
Getting JWSError JWSInvalidSignature using service_role

### DISCUSSION THREAD

**Discussion by warren-gallagher** [supabase/supabase#6123] [Questions] [answered]
After having been notified that my service_role key had leaked (Thanks for that), I updated my application with updated:
anon key, jwt secret & service_role. Also restarted the Supabase Server. 

Now requests using the supabase client with my service role is returning: JWSError JWSInvalidSignature using service_role

I have verified that the keys used in my app match what is in the supabase API settings.

Any thoughts?

**Comment by warren-gallagher** [comment, marked as answer]
I may have this resolved...My app is based on SvelteKit and I believe that the environment variables are resolved at build-time so I had to force a total rebuild so that the .env changes were picked up.

**Reply by ethanfox** [reply]
I am having this same issue and cannot find a fix for it. What do you mean by a forced total rebuild?
